### PR TITLE
Fix Executor handler annotation resolution for postponed (__future__) annotations

### PR DESCRIPTION
**Pull request by moonbox3**
## Summary
- Resolve handler `message` and `ctx` annotations before validation/registration so `from __future__ import annotations` works for `Executor` handlers.
- Prefer `typing.get_type_hints()` and only fall back on expected resolution errors; re-raise unexpected exceptions.
- When falling back, resolve BOTH `message` and `ctx` string annotations via `resolve_type_annotation()` to prevent string keys in handler registration.

## Tests
- Added `python/packages/core/tests/workflow/test_executor_future_annotations.py` asserting:
  - raw signature annotations are strings under postponed evaluation
  - handler registration uses concrete type keys (e.g., `str`), not string keys like `'str'`
  - inferred ctx output/workflow output types are correct
  - forced `get_type_hints` failure path still resolves both message and ctx annotations

### CODE REVIEW

**Review by moonbox3 — COMMENTED** [review]
## Automated PR Review Round 1: CHANGES_REQUESTED

Issue: #1
The change to resolve postponed annotations via `typing.get_type_hints()` before handler registration is on the right track, but the PR has correctness and security/reliability gaps that must be addressed before merging. The new test module currently duplicates the same test cases (running twice), and the `get_type_hints()` exception handling/fallback behavior is both risky (may mask real bugs) and incomplete (misses common exception types). Additionally, the added GitHub Actions workflows use an unsafe token-in-URL clone pattern that should be replaced with a safer checkout approach.

Blocking issues (4):
- Remove duplicated tests in `python/packages/core/tests/workflow/test_executor_future_annotations.py` (currently defined both as class methods and module-level functions), as pytest
- Harden `_validate_handler_signature` error handling: do not broadly swallow `TypeError`/`AttributeError` from `get_type_hints()` in a way that can mask real annotation/namespace bu
- `get_type_hints()` can raise `ValueError` for unresolved/invalid hints; current exception handling omits it, which can cause runtime hard-failures for postponed-annotation scenario
- GitHub Actions workflows added in this PR use `git clone https://<token>@github.com/...` which exposes credentials via command-line/process args and increases leak risk; replace wi

Suggestions (3):
- Add/adjust tests to explicitly pin the intended behavior for non-NameError failures from `get_type_hints` (e.g., ensure `ValueError` falls back if that’s the goal, and ensure truly
- Consider whether `include_extras=True` is needed when calling `get_type_hints()` (only if the framework relies on `Annotated` metadata) and add a targeted test if applicable.
- Consider adding a test for nested qualnames (e.g., `Outer.Inner.handle`) if owner/qualname parsing is part of handler resolution logic.

File: python/packages/core/tests/workflow/test_executor_future_annotations.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+            @handler
+            async def handle(self, message: str, ctx: WorkflowContext[FutureTypeA]) -> None:
+                pass
+
+        raw = inspect.signature(FallbackExecutor.handle).parameters
+        assert isinstance(raw["message"].annotation, str)
+        assert isinstance(raw["ctx"].annotation, str)
+
+        executor = FallbackExecutor(id="future_fallback")
+
```
Comment (moonbox3) [inline python/packages/core/tests/workflow/test_executor_future_annotations.py:98]: This file defines the same test cases twice (class-based tests and repeated module-level functions). Please delete one set to avoid duplicate execution and confusing double-failures; then use the space to add missing edge-case assertions instead.

File: python/packages/core/tests/workflow/test_executor_future_annotations.py [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+def test_handler_future_annotations_fallback_when_get_type_hints_fails(monkeypatch: pytest.MonkeyPatch) -> None:
+    import agent_framework._workflows._executor as executor_mod
+
+    def _raise_name_error(*_args, **_kwargs):
+        raise NameError("forced")
+
+    monkeypatch.setattr(executor_mod, "get_type_hints", _raise_name_error)
+
```
Comment (moonbox3) [inline python/packages/core/tests/workflow/test_executor_future_annotations.py:166]: These module-level tests duplicate the earlier class-based tests; keeping both will run everything twice under pytest. Remove this duplicate block.

File: python/packages/core/tests/workflow/test_executor_future_annotations.py [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+        assert isinstance(raw["ctx"].annotation, str)
+
+        executor = FutureUnionExecutor(id="future_union")
+        assert str in executor._handlers
+        assert "str" not in executor._handlers
+
+        assert set(executor.output_types) == {FutureTypeA, FutureTypeB}
+        assert executor.workflow_output_types == []
+
+    def test_handler_future_annotations_unparameterized_context(self) -> None:
```
Comment (moonbox3) [inline python/packages/core/tests/workflow/test_executor_future_annotations.py:61]: The fallback-path test only forces NameError. Add coverage for other `get_type_hints` failures (e.g., ValueError) and assert the intended behavior (fallback vs re-raise) so semantics don’t silently drift.

File: python/packages/core/agent_framework/_workflows/_executor.py [hunk 4]
```diff
@@ -717,25 +717,61 @@ def _validate_handler_signature(
...
+        owner_name = qualname.split(".")[-2]
+        owner = func.__globals__.get(owner_name)
+        if isinstance(owner, type):
+            localns = dict(vars(owner))
+
+    resolved_annotations: dict[str, Any] | None = None
+    try:
+        resolved_annotations = get_type_hints(func, globalns=func.__globals__, localns=localns)
+    except (NameError, TypeError, AttributeError):
+        resolved_annotations = None
```
Comment (moonbox3) [inline python/packages/core/agent_framework/_workflows/_executor.py:737]: Catching TypeError/AttributeError here can mask real handler annotation/namespace bugs and silently fall back to string resolution. Tighten the exception policy to only suppress expected resolution failures (and re-raise unexpected ones) so mis-annotated handlers fail fast instead of registering under wrong types.

File: python/packages/core/agent_framework/_workflows/_executor.py [hunk 5]
```diff
@@ -717,25 +717,61 @@ def _validate_handler_signature(
...
+
+    message_annotation: Any = message_param.annotation
+    ctx_annotation: Any = ctx_param.annotation
```
Comment (moonbox3) [inline python/packages/core/agent_framework/_workflows/_executor.py:740]: `typing.get_type_hints()` commonly raises ValueError for unresolved forward refs/invalid hints; the current except list omits it, so some postponed-annotation cases may hard-fail. Either include ValueError as an expected fallback trigger (with a test) or explicitly decide to fail fast and test that behavior.

File: python/packages/core/tests/workflow/test_executor_future_annotations.py (hunk 1 above)
Comment (moonbox3) [reply python/packages/core/tests/workflow/test_executor_future_annotations.py:98]: Automated status (r1): queued for remediation in this pass.

File: python/packages/core/tests/workflow/test_executor_future_annotations.py (hunk 2 above)
Comment (moonbox3) [reply python/packages/core/tests/workflow/test_executor_future_annotations.py:166]: Automated status (r1): queued for remediation in this pass.

File: python/packages/core/tests/workflow/test_executor_future_annotations.py (hunk 3 above)
Comment (moonbox3) [reply python/packages/core/tests/workflow/test_executor_future_annotations.py:61]: Automated status (r1): queued for remediation in this pass.

File: python/packages/core/agent_framework/_workflows/_executor.py (hunk 4 above)
Comment (moonbox3) [reply python/packages/core/agent_framework/_workflows/_executor.py:737]: Automated status (r1): queued for remediation in this pass.

File: python/packages/core/agent_framework/_workflows/_executor.py (hunk 5 above)
Comment (moonbox3) [reply python/packages/core/agent_framework/_workflows/_executor.py:740]: Automated status (r1): queued for remediation in this pass.
